backend/src/main/java/com/bakdata/conquery/models/events/CBlock.java

```diff
@@ -280,16 +297,19 @@ public static long calculateBitMask(int pathIndex, int[] mostSpecificChild) {
 				continue;
 			}
 
-			for (BucketEntry entry : bucket.entries()) {
-				if (!bucket.has(entry.getEvent(), column)) {
-					continue;
-				}
+			for (int entity : bucket.getEntities()) {
+				final int index = bucket.getEntityIndex(entity);
+				final int end = bucket.getEntityEnd(entity);
 
-				CDateRange range = bucket.getAsDateRange(entry.getEvent(), column);
+				for (int event = bucket.getEntityStart(entity); event < end; event++) {
+					if (!bucket.has(event, column)) {
+						continue;
+					}
 
-				final int index = bucket.getEntityIndex(entry.getEntity());
+					CDateRange range = bucket.getAsDateRange(event, column);
+					spans[index] = spans[index].spanClosed(range);
```

Comment:
Hier werden viele DateRanges kurzzeitig generiert. Du kannst Teile von CDateRange.spanClosed in Util-Methoden auslagern.
CDateRange::closedMin(ds1,ds1) -> int.

Ich weil nicht wie die vm optimiert, aber dann könntest du min und max immer im Register halten.

Reply:
Habe es ganz unrolled, schauen wa mal :)